verifier/verifier.go

```diff
@@ -59,7 +59,8 @@ func Verify(layout *Layout, attestations map[string]*dsse.Envelope, parameters m
 
 		acceptedKeys, err := envVerifier.Verify(context.Background(), env)
 		if err != nil {
-			return err
+			log.Infof("Unable to verify %s's signatures", attestationName)
```

Comment:
Looks safe... I think. As long as we handle all of the following cases, we will be OK:
1. Attestations for known steps
    1. Attestations for known keys
        1. Crossing the threshold
    3. Attestations for unknown keys
2. Attestations for unknown steps
    1. Attestations for known keys
    2. Attestations for unknown keys

I don't like mixing 1.1 and 2.1 as a preloaded check here. IMHO, we shouldn't even be considering (2) at all, but this requires a deeper refactoring.

Reply:
Deeper refactoring is a different question. In this verifier, I explicitly set out to differentiate claim authentication and claim verification. So all claims are loaded and authenticated into a format of "X says C for step S", and policy verification is for the question "does one of {X, Y, Z} say C for step S?"

Reply:
I'm open to the other approach where we use the policy to find what attestations we load and authenticate. The approach I was using seemed to also line up with having, say, a bundle of attestations that weren't on disk or so.

Reply:
Interesting discussion! I can see advantages to both approaches (using the policy to find attestations vs separate auth from verification). The former more tightly binds the attestations to the policy, and avoids checking irrelevant attestations, though I don't think that precludes retrieving attestations from a remote location. That info could be added to the layout. 

What I think is nice about the latter approach (the current design) is that it does give us more flexibility in terms of policy expression and verification. It lends itself more to expressing policy as "I want n of m trusted attestors to claim C about subject A during step S" and speculatively check an entire (set of) bundle. I'm trying to see a security risk in trying to _authenticate_ attestations for unknown signers (and steps for that matter), but can't immediately think of anything that jumps out right away.

Reply:
nb: we're not authenticating unknown signers after all, because the layout determines known signers.